wiki/Building-From-Source.md

```diff
@@ -1,96 +1,156 @@
+* Gentoo:
+  * https://wiki.gentoo.org/wiki/Android
+* Arch:
+  * https://wiki.archlinux.org/title/Android
+* Debian:
+  * https://wiki.debian.org/AndroidStudio
+* Ubuntu:
+  * https://help.ubuntu.com/community/AndroidSDK
+* Pop!\_OS:
+  * N/A
+
+The three mentioned developer applications can be installed from upstream; although the packages and setup responsible for the required tools can differ between distros, being:
+
+* **Arch**
+  * Packages can vary, read up on the Arch Wiki's [Android](https://wiki.archlinux.org/title/Android) page.
+
+* **Gentoo**
+  * `dev-util/android-studio`
+  * `dev-util/android-sdk-update-manager`
+  * `dev-util/android-ndk >= 25.1`
 
+For Debian, it requires to have the `non-free` repository to be enabled:
+* **Debian 12 / Ubuntu 22.10 / Pop!\_OS 22.10**
+  ```bash
+  sudo apt install android-sdk-platform-tools-common sdkmanager google-android-ndk-r25b-installer
+  ```
+
+## 2. Setting environment variables
+For Windows, set the environment variables:
+* `JAVA_HOME`:
+  * Example: `C:\Program Files\Android\Android Studio\jre`
+* `ANDROID_HOME`:
+  * Example: `%LOCALAPPDATA%\Android\Sdk`
+* `ANDROID_NDK_HOME`:
+  * Example: `%LOCALAPPDATA%\Android\Sdk\ndk\25.1.8937393`
+
+For Linux, the correct directories for the environment variables can greatly differ depending on the type of install. See the wiki page of your distro for more information:
+* Gentoo:
+  * https://wiki.gentoo.org/wiki/Android
+* Ubuntu:
+  * https://help.ubuntu.com/community/AndroidSDK#Post-Installation_Configuration
+
+Distro wikis that weren't listed above does not mention of environment variables, although generally they would be as:
+* `JAVA_HOME`:
+  * `/usr/lib/jvm/default-java/bin`
+* `ANDROID_HOME`:
+  * Arch: `~/Android/Sdk`
+  * Gentoo: `~/Android`
+  * Debian / Ubuntu / Pop!\_OS: `~/AndroidSDK`
+* `ANDROID_NDK_HOME`:
+  * Arch: `/opt/android-sdk/ndk`
+  * Linux: `/usr/lib/android-sdk/ndk`
+
+## 3. Building
 First you need to gather some additional resources in preparation for the build.  
 Move to the root directory of the project, then run this command:
 
 ```bash
 cargo xtask prepare-deps --platform android
 ```
 
-Next up is the proper build of the client. Run the following:
+Before building the client, Android has to have us to agree to the licenses otherwise building the client will halt and fail. To accept the agreements, follow the instructions for your corresponding OS:
+* Windows:
+  ```shell
+  cd "%ANDROID_SDK_ROOT%\tools\bin"
+  sdkmanager.bat --licenses
+  ```
+* Linux:
+  ```bash
+  cd ~/AndroidSDK
+  sdkmanager --licenses
+  ```
 
+Next up is the proper build of the client. Run the following:
 ```bash
 cargo xtask build-client --release
 ```
 
 The built APK will be in `build/alvr_client_quest`. You can then use adb or SideQuest to install it to your headset.
 
-To build and run:
-
+To build and run an apk for OpenXR as well, run:
```

Comment:
Don't change this. You don't need to specify it's for OpenXR since it's the only client build available.

Reply:
Doesn't that mean that section should replace the `cargo xtask build-client --release` section?

Reply:
Done!